### DISCUSSION TITLE
Confused about the thermostat behaviour

### DISCUSSION THREAD

**Discussion by 2fst4u** [jmcollin78/versatile_thermostat#472] [General]
Hi, so far this is awesome, it's just what I'm looking for.

I'm a bit confused about how the thermostat behaves when using it with another climate entity (in my case a heat pump).

When I turn off the versatile thermostat, my heat pump also turns off which I would expect. When I turn on the versatile thermostat, my heat pump turns on too. Great!

But when both are off and I go to turn on the heat pump manually, either in home assistant, or from its own app, or with the physical remote, then the versatile thermostat also turns on and starts interfering with the settings. Is there any way to prevent this? If the heat pump is manually turned on I'd like the versatile thermostat to remain off so that the manual settings can be used. The reason for this is not everyone in my household has the need or desire to use home assistant to control the settings and if they manually turn on the heat pump then they'd prefer to just have that manual setting remain what it is set to.

Is this possible at all?

**Comment by jmcollin78** [comment]
Hello @2fst4u ,

> But when both are off and I go to turn on the heat pump manually, either in home assistant, or from its own app, or with the physical remote, then the versatile thermostat also turns on and starts interfering with the settings. Is there any way to prevent thisc

This is normal. What we want is that VTherm follows the state on the underlying climate. We want that because many people want to continue to interact directly with the device without having to open HA. So the VTherm is listening to the underlying state and adapts itself to the last known state. There is no way to prevent this as it is a core feature of the 'over climate' VTherm type.

What you can do is set the preset to manual (no preset in fact) so that you can interact directly with your heatpump without interference of the VTherm.

**Reply by 2fst4u** [reply]
In manual mode the vtherm still adjusts the set point though? I just had it in that mode and it was adjusting to maintain the set temperature. 

On the first point though, if the underlying climate entity is manually adjusted with the remote, should the vtherm then match that setting and then continue to adjust to that new value from then on?

**Reply by jmcollin78** [reply]
> In manual mode the vtherm still adjusts the set point though? I just had it in that mode and it was adjusting to maintain the set temperature.

No. In manual mode, you set the setpoint (manually...)

> should the vtherm then match that setting and then continue to adjust to that new value from then on

When you set the setpoint with the remote, then the VTherm switch to manual mode (ie no preset) automatically. The use of the remote means "the user wants to manual set the target temp and don't to use a preset anymore".

I hope it is more clear now.

**Reply by 2fst4u** [reply]
Hmm I wish it was more clear but I'm still a bit confused sorry. That behaviour is what I would hope would happen which is great if that's what should be happening, but as I mentioned I have vtherm in manual mode and it's making adjustments to meet the target temperature. What could be going on there?

**Reply by jmcollin78** [reply]
Maybe you have regulation activated ?

In manual mode you set the target setpoint but the regulation will increase or decrease the setpoint to reach the target. This is the only explanation I see.

**Reply by 2fst4u** [reply]
I do have regulation set, that will be why. But I'm confused about why you would use vtherm without using regulation. Isn't that the point of vtherm?

If I turn off regulation, what is the advantage to using vtherm? How would I use it to achieve the intent of maintaining a target temperature?

Pure curiosity, I just want to make sure I'm using it correctly and not misunderstanding.

**Reply by jmcollin78** [reply]
Without regulation you have all the features: window detection, movement, power management, ...

In onver_climate as you are the regulation is not mandatory. The underlying device do it normally. You need regulation if the underlying device does a bad regulation (which is often the case).

**Reply by 2fst4u** [reply]
Yea my heat pump has poor regulation which is why I want to use the regulation feature.

In that case, is there any way I can achieve what I want out of the system? Can manual mode have regulation disabled while other modes keep regulation?

If not, the only way I can achieve it is to disable the integration when I'm not using it which I really don't want to do!

**Reply by jmcollin78** [reply]
> In that case, is there any way I can achieve what I want out of the system? Can manual mode have regulation disabled while other modes keep regulation?

That is not possible now. I understand your point but I will not change the behavior: manual mode means "set setpoint manually" and not "deactivate auto-regulation".

but what I think is that you will never succeed to have 2 sources of truth: the device itself or VTherm. You have to choose, if you want use VTherm or not. You cannot choose to use one part of Vtherm combined with manual operations. That will never work fine.
Why just let VTherm do its job ? Is that not a habit you have to regulate yourself by changing the target ? Set a setpoint (manually or via preset) and let the system do its job. You will forget it after a while.

**Reply by 2fst4u** [reply]
Not everyone in my house is on board with using the features I like to implement unfortunately, so I was hoping to be able to leave the regulation in an off state until I personally want it enabled.

Wouldn't it make sense for manual mode to disable regulation though? I understand if it's something you don't want to change because it's your project after all. 

Thanks for explaining.

**Reply by jmcollin78** [reply]
> Wouldn't it make sense for manual mode to disable regulation though?

No. Manual means setpoint is set manual and not by preset. Not more.

> Not everyone in my house is on board with using the features I like to implement

The only think needed is to set the target. With all the integrated features (presence, movement, ...) you will never have to change manually the target. In my house, I never change the setpoint for 2 years. I have automated the preset changes with scheduler and that's it. No one have to change something. That is the correct to use VTherm.

**Reply by jmcollin78** [reply]
Nevertheless, it is a very important discussion on the VTherm behavior.  Thank you for this !

**Reply by chriske** [reply]
@jmcollin78 and what about temporary disabling the auto regulation? My AC units beeping all the time when the regulator changes the target temperature, and the beeping can't be disabled unfortunately. Currently I'm disabling the regulation manually every evning, and enabling it after we wake up (for 3 AC units). It would be nice to be able to do this automatically somehow.

**Reply by 2fst4u** [reply]
Even disabling regulation doesn't really help because the last temperature setting of vtherm still overrides the underlying climate entity's set point. For example if it was set to 16 and it gets turned off, then someone manually turns on the heat pump to 14, the vtherm setting will just change the heat pump to 16.

I might be wrong but I think that's what I just observed when I tried turning off the regulation temporarily for a test. Overnight my heat pump automatically kicked in because it got too cold, and it turned on to 12, but when I woke up it was set to 16 which is what vtherm was set to.

It's a bit of a shame, I'd really like to use this integration because the regulation is just what I was looking for but the inability to let manual settings override the vtherm settings is holding me back sorry.

Even if manually changing the underlying climate entity temperature forced vtherm to follow that setting I'd be happy. If using the remote meant vtherm then started regulating to the new value intended by the manual change that would be acceptable to other people in my house.

**Reply by 2fst4u** [reply]
@jmcollin78 I see in another issue it's mentioned that the vtherm temperature should change when you change the underlying thermostat temperature, e.g. with the heat pump remote. Is that correct? Because that doesn't seem to be working for me.

**Reply by jmcollin78** [reply]
Yes this is designed like this: if you change directly on the AC with the remote for example, the VTherm should go into manual mode (no preset) and target should be copied.

**Reply by 2fst4u** [reply]
There appears to be an issue there because this isn't happening. I don't know if it's just me or not. When I use the remote the vtherm target doesn't follow the manual setting. If this worked then I'd be happy with the behaviour of regulation.

**Reply by 2fst4u** [reply]
Should I open an issue to track debugging of this?

**Reply by jmcollin78** [reply]
I will dot it for you. I will do some tests with mine to try to reproduce. Maybe something have been broken.

**Comment by jmcollin78** [comment]
Fixed by https://github.com/jmcollin78/versatile_thermostat/releases/tag/6.3.4.beta5
